dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/PickleUtils.cs

```diff
@@ -288,7 +284,33 @@ public static string rawStringFromBytes(byte[] data)
         }
         return result;
 	}
-	
+
+    internal static int CheckedCast(long length)
+    {
+        try
+        {
+            return checked((int)length);
+        }
+        catch (OverflowException x)
+        {
+            throw new PickleException("pickle too large, can't read more than maxint", x);
+        }
+    }
+
+    internal static string GetStringFromUtf8(in ReadOnlySpan<byte> utf8)
+    {
+        if (utf8.IsEmpty)
+            return string.Empty;
+
+        unsafe
+        {
+            fixed (byte* bytes = utf8)
+            {
+                return Encoding.UTF8.GetString(bytes, utf8.Length);
```

Comment:
There's no netstandard way to do this without unsafe ?

Reply:
Arrow has the same needs:

https://github.com/apache/arrow/blob/044b418fa108a57f0b4e2e887546cc3e68271397/csharp/src/Apache.Arrow/Arrays/StringArray.cs#L46-L47

I haven't found a way without `unsafe` code.

Reply:
@grabyourpitchforks ?

Reply:
The new OperationStatus-based UTF-8 encoder is in .NET Core 3.0. We end up with a similar concern within Json: https://github.com/dotnet/corefx/blob/8d2714b8af06bd563a2015c1b3155f04a91a4b1a/src/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs#L110-L139

Reply:
Right, _unsafe_ is the only way to do this with `ROS<T>` for now unless you're willing to make a temporary copy of the input into its own `T[]`. In theory we could ship a "netstandard compat" package that adds this and a bunch of other APIs as extension methods. I don't know if there were ever any plans to do that, though.

Reply:
Fine. I was most interested that there was an API in .NET Core 3.0 so long term we're in a good place.

Reply:
@danmosemsft I found two  other things missing in .NET Standard 2.0:

- possibility to parse an int/double/long/bool from `Span<byte>` which iternally represents a `string` (not `utf8string`). I know that `int.Parse` etc were added to the types directly, not as extension methods so we can't add them to `int` to Full .NET Framework. But this is a common scenario when writing .NET Standard 2.0 performant code. 
- possibility to parse `BigInteger` from `Span<byte>`

Reply:
> I found two other things missing in .NET Standard 2.0

This is all by design.

Reply:
> This is all by design.

I am aware of that but it creates a situation where we recommend the library authors to target .NET Standard for libs which makes it impossible to use some of the perf gains introduced in Core 2.1/2.2/3.0

Reply:
That's just the way things are. We are not going to duplicate all new functionality in .NET Core into downlevel libraries.